library/core/src/num/dec2flt/mod.rs

```diff
@@ -204,155 +194,69 @@ impl ParseFloatError {
     }
 }
 
-#[stable(feature = "rust1", since = "1.0.0")]
 impl fmt::Display for ParseFloatError {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         self.__description().fmt(f)
     }
 }
 
-fn pfe_empty() -> ParseFloatError {
+pub fn pfe_empty() -> ParseFloatError {
```

Comment:
You have a whole lot more of `pub fn` now. That doesn't seem right, these implementation details aren't part of the public API. If you just need to be able to use them in other parts of dec2flt, can you make them `pub(super)` instead?

That would also give compiler warnings for unused functions, like @the8472 mentioned below.

Reply:
I've removed most of them, except for `pfe_invalid`, for unittesting reasons. Also, `#![feature(dec2flt)]` needs to be enabled to use these, which requires nightly and should make it pretty inaccessible to most crates (unless this [cursed](https://twitter.com/m_ou_se/status/1410930900116951040) but beautiful hack is used in production).

It's worth noting the old implementation exposed:

- parse
   - Decimal
   - ParseResult
   - parse_decimal
- rawfp 
  - Unpacked
  - RawFloat
  - fp_to_float
  - round_normal
  - encode_normal
  - encode_subnormal
  - prev_float
  - next_float

Meanwhile, my implementation exposes (after I commit my latest updates):

- float
  - RawFloat
- lemire
  - compute_float
- number
  - Number
- parse
  - parse_number
  - parse_inf_nan
- pfe_invalid

All of which I think are justified for unit tests.

Reply:
Changes in the last commit include:

- pr_empty `pub` => `pub(super)
- common `pub mod` => `mod`
- biased_fp_to_float `pub fn` => `fn`
- INT_POW10 `pub const` => `const`
- parse_partial_number `pub fn` => `fn`
- parse_partial_inf_nan `pub fn` => `fn`